codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydotnet/TypeConversionCodegen.java

```diff
@@ -1006,7 +1007,7 @@ protected TokenTree errorFromDanyBody(final TreeSet<StructureShape> errorShapes)
             }
 
             if (dependentNamespaces.size() > 0) {
-                Set<TokenTree> cases = new HashSet<>();
+                List<TokenTree> cases = new ArrayList<>();
```

Comment:
Changing this from a Set to List may allow for duplicates.
I think that is currently prevented by `dependentNamespaces` being a Set.
But is there an ordered set we could use instead?

Reply:
Yeah I tried `TreeSet` at first but then realized `TokenTree` wasn't comparable. I can definitely avoid the list.

Reply:
We could find a Set that preserves insertion ordering.
Or we can add a comment:
```suggestion
                // dependentNamespace is collision free, so cases will be collision free
                List<TokenTree> cases = new ArrayList<>();
```

Smithy-Dafny has lots of authors modifying logic all over the place.
I am trying to remove or document any logic that assumes the behavior of other logic.

Reply:
For the record the `dependentNamespaces` source is already a `TreeSet`, so the way the code is structured now, duplicates are still impossible. But I changed to use a Stream anyway since it simplified the code too.